lua/entities/gmod_wire_expression2/core/e2lib.lua

```diff
@@ -439,89 +439,135 @@ end
+				E2Lib.extensions.list[ #E2Lib.extensions.list + 1 ] = row.name
+				sortExtensionsList = true
+			end
+		end
 	end
 
-	function extensions.SetStatus(name, status)
-		sql.Query(string.format("REPLACE INTO wire_expression2_extensions (name, enabled) VALUES (%s, %d)", sql.SQLStr(name), status and 1 or 0))
+	function E2Lib.SetExtensionStatus( name, status )
+		E2Lib.extensions.status[ name ] = status
+		sql.Query( string.format( "REPLACE INTO wire_expression2_extensions (name, enabled) VALUES (%s, %d)", sql.SQLStr( name ), status and 1 or 0 ) )
 	end
 
-	function E2Lib.RegisterExtension(name, default)
-		local status = extensions.GetStatus(name, default)
-		if not status then Error("Skipping disabled E2 extension '" .. name .. "'. To enable, run 'wire_expression2_extension_enable " .. name .. "'\n") end
+	--[[ useless yet
+	function E2Lib.GetExtensionStatus( name )
+		return E2Lib.extensions.status[ name ] or false
 	end
-
-	concommand.Add("wire_expression2_extension_enable", function(ply, cmd, args)
-		if ply:IsValid() then
-			if not ply:IsSuperAdmin() and not game.SinglePlayer() then return end
+	]]
+	
+	function E2Lib.GetExtensions( niceList )
+		if sortExtensionsList then
+			table.sort( E2Lib.extensions.list, function( a, b ) return a < b end )
+			sortExtensionsList = nil
 		end
-
-		if extensions.GetStatus(args[1]) then
-			if ply:IsValid() then
-				ply:PrintMessage(2, "Extension '" .. args[1] .. "' is already enabled. Did you remember to reload Expression 2 using the console command 'wire_expression2_reload'?")
-			else
-				print("Extension '" .. args[1] .. "' is already enabled. Did you remember to reload Expression 2 using the console command 'wire_expression2_reload'?")
-			end
-			return
+		
+		if not niceList then return E2Lib.extensions.list end
+		
+		local l = 0
+		local t = {}
+		for _, n in ipairs( E2Lib.extensions.list ) do if n:len() > l then l = n:len() end end
+		for k, n in ipairs( E2Lib.extensions.list ) do t[k] = n .. string.rep( " ", l - n:len() ) .. "   " .. ( E2Lib.extensions.status[n] and "enabled" or "disabled" ) end
+		return t
+	end
+	
+	function E2Lib.RegisterExtension( name, default )
+		if E2Lib.extensions.status[ name ] == nil then
+			E2Lib.SetExtensionStatus( name, default or false )
+			E2Lib.extensions.list[ #E2Lib.extensions.list + 1 ] = name
+			sortExtensionsList = true
 		end
-
-		extensions.SetStatus(args[1], true)
-
-		if extensions.GetStatus(args[1]) then
-			if ply:IsValid() then
-				ply:PrintMessage(2, "Extension '" .. args[1] .. "' enabled. Now reload Expression 2 using the console command 'wire_expression2_reload'.")
-			else
-				print("Extension '" .. args[1] .. "' enabled. Now reload Expression 2 using the console command 'wire_expression2_reload'.")
-			end
+		
+		-- DO NOT MODIFY THIS LINE, PLEASE!
+		assert( E2Lib.extensions.status[ name ], "Skipping disabled E2 extension '" .. name .. "'. To enable, run 'wire_expression2_extension_enable " .. name .. "'\n" )
```

Comment:
This comment should probably be more clear lol.

So this is the new 'error' function? I wonder if we need to switch all our errors to asserts to get around the RunString bug Divran uncovered in #456